homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,236 @@
+https://home-assistant.io/components/sensor.seventeentrack/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, ATTR_LOCATION, CONF_PASSWORD, CONF_SCAN_INTERVAL,
+    CONF_USERNAME)
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle, slugify
+
+REQUIREMENTS = ['py17track==2.0.1']
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DESTINATION_COUNTRY = 'destination_country'
+ATTR_INFO_TEXT = 'info_text'
+ATTR_ORIGIN_COUNTRY = 'origin_country'
+ATTR_PACKAGE_TYPE = 'package_type'
+ATTR_TRACKING_INFO_LANGUAGE = 'tracking_info_language'
+
+CONF_SHOW_ARCHIVED = 'show_archived'
+CONF_SHOW_DELIVERED = 'show_delivered'
+
+DATA_PACKAGES = 'package_data'
+DATA_SUMMARY = 'summary_data'
+
+DEFAULT_ATTRIBUTION = 'Data provided by 17track.net'
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=10)
+
+VALUE_DELIVERED = 'Delivered'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SHOW_ARCHIVED, default=False): cv.boolean,
+    vol.Optional(CONF_SHOW_DELIVERED, default=False): cv.boolean,
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from py17track import Client
+    from py17track.errors import SeventeenTrackError
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+
+    client = Client(websession)
+
+    try:
+        login_result = await client.profile.login(
+            config[CONF_USERNAME], config[CONF_PASSWORD])
+
+        if login_result is False:
+            _LOGGER.error('Invalid username and password provided')
+            return False
+    except SeventeenTrackError as err:
+        _LOGGER.error('There was an error while logging in: %s', err)
+        return False
+
+    if CONF_SCAN_INTERVAL in config:
+        scan_interval = config[CONF_SCAN_INTERVAL]
+    else:
+        scan_interval = DEFAULT_SCAN_INTERVAL
+
+    data = SeventeenTrackData(
+        client, scan_interval, config[CONF_SHOW_ARCHIVED],
+        config[CONF_SHOW_DELIVERED])
+    await data.async_update()
+
+    sensors = []
+
+    for status, quantity in data.summary.items():
+        sensors.append(SeventeenTrackSummarySensor(data, status, quantity))
+
+    for package in data.packages:
+        sensors.append(SeventeenTrackPackageSensor(data, package))
+
+    async_add_entities(sensors, True)
+
+
+class SeventeenTrackSummarySensor(Entity):
+    """Define a summary sensor."""
+
+    def __init__(self, data, name, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._name = name
```

Comment:
Isn't this more of a type than a name? It's different status of packages right?

Reply:
You got it. I'll change the attribute to be more descriptive.